app/models/course/video.rb

```diff
@@ -95,4 +97,11 @@ def initialize_duplicate_tab(duplicator, other)
                  duplicator.options[:target_course].video_tabs.first
                end
   end
+
+  def set_duration
+    return if duration
```

Comment:
Since we allow the video URL to be modified, I think we should take care of the case when it is. Perhaps we could move this check out of `set_duration` and have all the logic that decides whether to update `duration` as a condition on `before_validation`?

Reply:
I think that is a mistake on it's own right. Changing video URL actually breaks a ton of other things. I'll add a validation to prevent that and update the UI.